src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3487,45 +3489,45 @@ void Arguments::extract_shared_archive_paths(const char* archive_path,
   len = end_ptr - begin_ptr;
   cur_path = NEW_C_HEAP_ARRAY(char, len + 1, mtInternal);
   strncpy(cur_path, begin_ptr, len + 1);
-  //cur_path[len] = '\0';
   FileMapInfo::check_archive((const char*)cur_path, false /*is_static*/);
   *top_archive_path = cur_path;
 }
 
-bool Arguments::init_shared_archive_paths() {
+void Arguments::init_shared_archive_paths() {
   if (ArchiveClassesAtExit != NULL) {
```

Comment:
Maybe change `NULL` to `nullptr` since your new code below in the same function uses `nullptr`?